src/test/functionsDeployHelper.spec.ts

```diff
@@ -284,4 +295,453 @@ describe("functionsDeployHelper", () => {
+        ],
+        functionsToDelete: [],
+        schedulesToDelete: [],
+      };
+      expect(deploymentPlan).to.deep.equal(expected);
+    });
+
+    it("should put existing functions being deployed into functionsToUpdate", () => {
+      const regionMap: helper.RegionMap = {
+        "us-east1": [
+          {
+            name: "projects/a/locations/us-east1/functions/c",
+            labels: {},
+            environmentVariables: {},
+            entryPoint: "",
+          },
+          {
+            name: "projects/a/locations/us-east1/functions/d",
+            labels: {},
+            environmentVariables: {},
+            entryPoint: "",
+          },
+        ],
+        "us-west1": [
+          {
+            name: "projects/a/locations/us-west1/functions/d",
+            labels: {},
+            environmentVariables: {},
+            entryPoint: "",
+          },
+        ],
+      };
+      const existingFunctions: helper.CloudFunctionTrigger[] = [
+        {
+          name: "projects/a/locations/us-east1/functions/c",
+          labels: {},
+          environmentVariables: {},
+          entryPoint: "",
+        },
+        {
+          name: "projects/a/locations/us-east1/functions/d",
+          labels: {},
+          environmentVariables: {},
+          entryPoint: "",
+        },
+        {
+          name: "projects/a/locations/us-west1/functions/d",
+          labels: {},
+          environmentVariables: {},
+          entryPoint: "",
+        },
+      ];
+      const filters: string[][] = [];
+
+      const deploymentPlan = helper.createDeploymentPlan(regionMap, existingFunctions, filters);
+
+      const expected: helper.DeploymentPlan = {
+        regionalDeployments: [
+          {
+            region: "us-east1",
+            functionsToCreate: [],
+            functionsToUpdate: [
+              {
+                name: "projects/a/locations/us-east1/functions/c",
+                labels: {},
+                environmentVariables: {},
+                entryPoint: "",
+              },
+              {
+                name: "projects/a/locations/us-east1/functions/d",
+                labels: {},
+                environmentVariables: {},
+                entryPoint: "",
+              },
+            ],
+            schedulesToCreateOrUpdate: [],
+          },
+          {
+            region: "us-west1",
+            functionsToCreate: [],
+            functionsToUpdate: [
+              {
+                name: "projects/a/locations/us-west1/functions/d",
+                labels: {},
+                environmentVariables: {},
+                entryPoint: "",
+              },
+            ],
+            schedulesToCreateOrUpdate: [],
+          },
+        ],
+        functionsToDelete: [],
+        schedulesToDelete: [],
+      };
+      expect(deploymentPlan).to.deep.equal(expected);
+    });
+
+    it("should delete existing functions not in local code, only if they were deployed via CLI", () => {
```

Comment:
Just checking that you didn't do anything to alter the logic that prompts before doing any deletes.

Reply:
Nope (or at least I didn't intend to). This may have just been a confusing title for the test - basically, this is checking that when there are no --only filters, we delete all existing functions that a - have the deployment-tool:cli-firebase label and b- don't have a matching local function.